internal/command/machine/status.go

```diff
@@ -46,6 +47,29 @@ func newStatus() *cobra.Command {
 	return cmd
 }
 
+func getProcessGroup(m *api.Machine) string {
+	// backwards compatible process_group getter.
+	// from poking around, "fly_process_group" used to be called "process_group"
+	// and since it's a metadata value, it's like a screenshot.
+	// so we have 3 scenarios
+	// - machines with only 'process_group'
+	// - machines with both 'process_group' and 'fly_process_group'
+	// - machines with only 'fly_process_group'
+
+	processGroup := m.Config.Metadata["process_group"]
+	flyProcessGroup := m.Config.Metadata[api.MachineConfigMetadataKeyFlyProcessGroup]
+
+	if processGroup != "" && flyProcessGroup != "" {
+		return flyProcessGroup
+	}
```

Comment:
Temoving these lines will have the same effect because of below `if`.
Such a small function could use a quick test case added.